ships/0027-image-abstraction-layer.md

```diff
@@ -0,0 +1,233 @@
+---
+title: shipwright-image-abstraction-layer
+authors:
+  - "@ricardomaraschini"
+reviewers:
+  - "@ImJasonH"
+  - "@adambkaplan"
+  - "@GabeMontero"
+approvers:
+  - "@ImJasonH"
+  - "@adambkaplan"
+  - "@GabeMontero"
+creation-date: 2022-02-07
+last-updated: 2022-02-07
+status: implementable
+see-also:
+  - "/ships/0020-shipwright-image-api.md"  
+---
+# Shipwright Image Abstraction layer
+
+This enhancement proposal suggests an image management abstraction layer for a Kubernetes cluster
+in the form of an Operator; this abstraction layer, in some aspects, resembles OpenShift’s
+ImageStream. The idea behind this EP is to design the base API types needed to support such an
+abstraction as well as lay down a basic set of features aiming to have a minimal viable product
+to further validate the idea. As chosen set of features intended for the minimal viable product
+are:
+
+* The ability to create references by hash to Containers Images hosted in a remote registry.
+* The ability of mirroring remote images in an  “on-premise” image registry.
+
+## Motivation
+
+Keeping track of all Container Images in use in a Kubernetes cluster is a complicated task.
+Container Images may come from numerous different Image Registries. To add to this, Container
+Runtimes rely on remote registries (from the cluster's point of view) when obtaining Container
+Images, potentially making the process of pulling their blobs (manifests, config, and layers)
+slower.
+
+The notion of indexing Container Image versions by tags (“latest”, “development”, etc) is helpful.
+Still it does not provide users with the full confidence to always use the intended Container
+Image – today's "latest" tag might not be tomorrow's "latest" tag. In addition to that, these
+Image Registries allow access to Container Images by their Manifest content's hash (i.e., usually
+sha256), which gives users full confidence at a cost in semantics. Other factors may pop up when
+running, for instance, in an air-gapped environment, where the cluster nodes may not be allowed to
+reach any external Image Registry.
+
+This enhancement proposal suggests an image management abstraction layer. For instance, by
+providing a direct mapping between a Container Image tag (e.g., "latest") and its corresponding
+Manifest content's hash, users can then refer to the Container Image by its name – and be sure to
+use that specific version. 
+
+In summary, the proposed operator mirrors and indexes remote Container Images into a Kubernetes
+cluster, allowing for other Operators to easily navigate through these multiple indexed Images.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
```

Comment:
I would argue it's implementable, given the fact it's already :-)

Reply:
Done.